localstack/aws/handlers/auth.py

```diff
@@ -45,10 +45,7 @@ def __call__(self, chain: HandlerChain, context: RequestContext, response: Respo
         set_aws_access_key_id(access_key_id)
 
         # Obtain the account ID and save it in the request context
-        if account_id_from_header := context.request.headers.get(HEADER_LOCALSTACK_ACCOUNT_ID):
-            context.account_id = account_id_from_header
-        else:
-            context.account_id = get_account_id_from_access_key_id(access_key_id)
```

Comment:
Removed because this was the wrong assumption to make - that CSA requests happen only within the same account. Also see note below.